### ISSUE TITLE
as_factor () Error: `idx` must contain one integer for each level of `f`

### ISSUE THREAD

**Issue by antaldaniel** [tidyverse/forcats#89]
column <- c("hello", "dear", "hello", NA, "dear")
column <- forcats::as_factor (column)
Error: `idx` must contain one integer for each level of `f`

I think in this case three options could be considered:
1. convert NA values to a factor with level = "NA";
2. give a more meaningful error message;
3. or just leave NA values as they are, to be consistent with as.factor ()  which handles NA values.

I think the third option is the best, but I do not see what other behavior may  be considered.

**Comment by fabian-s** [comment]
seconded. this behavior makes the function near unusable for practical purposes....

**Comment by kvittingseerup** [comment]
I got the same error message when accidentally using `fct_inorder` on grouped data. Is that a desirable behavior?

**Comment by ChristianRohde** [comment]
Wow, I also got stuck on this and it took me a while to understand that the reason was NA in the column which was used to order the data. The problem was that I did not carefully use base::rowMeans() with na.rm = TRUE. As a result a function which worked long time with my data and somehow so far never included NA values suddenly died with a error I did not understand. I have to admit that I did not figure out the solution based on the error but rather needed Google to find this page to get into it. I would appreciate a meaningful error message too!
